add_cart_s-information.php

```diff
@@ -59,28 +59,28 @@
             if ($type === 'vip') {
                 $price *= 0.9; // vip user discount
             }
-            $orderid = $_SESSION['order_id']; 
+            $orderid = $order_id; 
 
             $sqlCheck = "SELECT num_dishes FROM order_detail WHERE order_id = :order_id AND dish_id = :dish_id"; 
             $stmtCheck = $conn->prepare($sqlCheck);
             $stmtCheck->bindParam(":order_id", $orderid, PDO::PARAM_INT); 
-            $stmtCheck->bindParam("dish_id", $dishId, PDO::PARAM_INT); 
+            $stmtCheck->bindParam(":dish_id", $dishId, PDO::PARAM_INT); 
             $stmtCheck->execute();
             $orderDetail = $stmtCheck->fetch(PDO::FETCH_ASSOC);
             if ($orderDetail) {
                 $numDishes = $orderDetail["num_dishes"]+1;
                 $sqlUpdate = "UPDATE order_detail SET num_dishes = :num_dishes, price = :price WHERE order_id = :order_id AND dish_id = :dish_id";
                 $stmtUpdate = $conn->prepare($sqlUpdate);
-                $stmtUpdate->bindParam("num_dishes", $numDishes, PDO::PARAM_INT); 
-                $stmtUpdate->bindParam("price", $price, PDO::PARAM_STR); 
-                $stmtUpdate->bindParam("order_id", $orderid, PDO::PARAM_INT); 
-                $stmtUpdate->bindParam("dish_id", $price, PDO::PARAM_INT); 
+                $stmtUpdate->bindParam(":num_dishes", $numDishes, PDO::PARAM_INT); 
+                $stmtUpdate->bindParam(":price", $price, PDO::PARAM_STR); 
+                $stmtUpdate->bindParam(":order_id", $orderid, PDO::PARAM_INT); 
+                $stmtUpdate->bindParam(":dish_id", $price, PDO::PARAM_INT); 
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Incorrect parameter binding for dish_id</b></div>
<div id="fix">

The parameter binding for `dish_id` appears to be using `$price` instead of `$dishId`. This could lead to incorrect data being updated in the database.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">


```suggestion
                $stmtUpdate->bindParam(":dish_id", $dishId, PDO::PARAM_INT);
```

</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/Lux-Jason/rms_new/pull/11#issuecomment-2558502994>#66901b</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged